axon-server-connector/src/main/java/org/axonframework/axonserver/connector/event/LegacyAxonServerEventStorageEngine.java

```diff
@@ -184,20 +185,17 @@ public void rollback() {
         });
     }
 
-    private void assertValidTags(List<TaggedEventMessage<?>> events, String aggregateType, String aggregateIdentifier) {
+    private static AtomicLong resolveSequencer(Map<String, AtomicLong> aggregateSequences, String aggregateIdentifier,
+                                               AggregateBasedConsistencyMarker consistencyMarker) {
+        return aggregateSequences.computeIfAbsent(aggregateIdentifier,
```

Comment:
I had to reason about this a bit: whether we really should return a position of `-1` if the `aggregateIdentifier` isn't part of the original `AppendCondition`. But, this is for the use case that a new aggregate was constructed within the scope of another aggregate.

Differently put, when `EventStoreTransaction#source` wasn't invoked, the `AppendCondition` couldn't reference the criteria of the `SourcingCondition`, and hence we assume its an event for a new aggregate (or, in new terms, `Tag`).

Although "I know you longer then today," hence I am pretty sure that's the reason, is still figured it valuable to drop a comment to verify this assumption. So, here it is!

Reply:
What put me off was the removal of the else-branch in the `assertValidTags` method, by the way. It used to validate a single tag, matching the discovered aggregate identifier and aggregate type. These, if kept, wouldn't merit a default position of -1 for events that don't have a matching tag in the `AppendCondition`, I think.

Reply:
I'm not sure that I completely follow the reasoning.

The appendCondition describes what the consumer has "seen", based on which a decision was made. This may be 1 or multiple aggregates. If an event is published with a tag that wasn't "seen" before, then we must insert that with sequence number 0. If the tag (/aggregate) already exists, the transaction will be rejected.

Reply:
No, we align Allard, thanks for clarifying :-)

Unrelated side note: I think "seen" is an assumption. Somebody might invoke the `EventStorageEngine` directly with whatever `AppendCondition` they'd like. Hence, that doesn't mean they had any previous sourcing operations impact the `AppendCondition` that's given here.